File: dotnet/DesignPatterns/CreationalDesignPatterns/PrototypeDesignPattern/PrototypeDesignPattern.Example1/Models/DocumentMetadata.cs
namespace PrototypeDesignPattern.Example1.Models;

public class DocumentMetadata
{
    public string Author { get; set; } = string.Empty;
    public string Version { get; set; } = "1.0";
    public List<string> Tags { get; set; } = new();
    public Dictionary<string, string> CustomProperties { get; set; } = new();
    public int PageCount { get; set; } = 1;

    public override string ToString() =>
        $"Author: {Author}, Version: {Version}, Pages: {PageCount}, Tags: [{string.Join(", ", Tags)}]";
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/PrototypeDesignPattern/PrototypeDesignPattern.Example1/src/models/DocumentMetadata.ts
export class DocumentMetadata {
  author: string = '';
  version: string = '1.0';
  tags: string[] = [];
  customProperties: Record<string, string> = {};
  pageCount: number = 1;

  toString(): string {
    return `Author: ${this.author}, Version: ${this.version}, Pages: ${this.pageCount}, Tags: [${this.tags.join(', ')}]`;
  }
}
